consensus/src/experimental/execution_phase.rs

```diff
@@ -75,24 +54,21 @@ impl ExecutionPhase {
 impl StatelessPipeline for ExecutionPhase {
     type Request = ExecutionRequest;
     type Response = ExecutionResponse;
-    async fn process(&self, req: ExecutionRequest) -> ExecutionResponse {
-        let ExecutionRequest { blocks } = req;
+    async fn process(&self, req: ExecutionRequest) -> ResponseWithInstruction<ExecutionResponse> {
+        let ExecutionRequest { ordered_blocks } = req;
 
         // execute the blocks with execution_correctness_client
-        let resp_inner = blocks
-            .into_iter()
+        let resp_inner = ordered_blocks
+            .iter()
             .map(|b| {
-                let state_compute_result = self
-                    .execution_proxy
-                    .compute(&b, b.parent_id())
-                    .map_err(|e| ExecutionPhaseError {
-                        block_id: b.id(),
-                        error: e,
-                    })?;
-                Ok(ExecutedBlock::new(b, state_compute_result))
+                let state_compute_result =
+                    self.execution_proxy.compute(b.block(), b.parent_id())?;
+                Ok(ExecutedBlock::new(b.block().clone(), state_compute_result))
             })
-            .collect::<Result<Vec<ExecutedBlock>, ExecutionPhaseError>>();
+            .collect::<Result<Vec<ExecutedBlock>, ExecutionError>>();
+
+        // TODO: empty the request channel non-blockingly as they will also fail
 
-        ExecutionResponse { inner: resp_inner }
+        ResponseWithInstruction::from(ExecutionResponse { inner: resp_inner })
```

Comment:
the Instruction is not used?

Reply:
yeah, I will add let it return Instruction::Clear in the next PR (we have a TODO above it)